iml-agent/src/action_plugins/high_availability/corosync_conf.rs

```diff
@@ -0,0 +1,257 @@
+    s: impl Iterator<Item = String>,
+    new_port: u16,
+) -> impl Iterator<Item = String> {
+    lazy_static! {
+        static ref RE: Regex = Regex::new(r"(\s*mcastport:\s*)\d+").unwrap();
+    }
+
+    s.map(move |x| {
+        if x.trim().starts_with("mcastport:") {
+            RE.replace(&x, |caps: &regex::Captures| {
+                format!("{}{}", &caps[1], new_port)
+            })
+            .to_string()
+        } else {
+            x
+        }
+    })
+}
+
+/// Given a new mcast port, updates the port on each interface,
+/// bumps the config_version and writes out the new corosync.conf.
+pub(crate) async fn change_mcast_port(new_port: u16) -> Result<(), ImlAgentError> {
+    let conf = read_corosync_conf().await?;
+    let conf = std::str::from_utf8(&conf)?
+        .trim()
+        .lines()
+        .map(|x| x.to_string());
+
+    let (totem, _) = corosync_cmapctl().await?;
+
+    let conf = update_config_version(conf, totem.next_version());
+    let conf = update_mcast_port(conf, new_port);
+    let conf = conf.collect::<Vec<_>>().join("\n");
+
+    write_corosync_conf(conf.as_bytes()).await?;
+
+    Ok(())
+}
+
+type StringMap<'a> = HashMap<&'a str, &'a str>;
+
+#[derive(Debug)]
+pub(crate) struct Totem {
+    config_version: Option<u64>,
+    cluster_name: String,
+    token: u32,
+}
+
+impl Totem {
+    fn next_version(&self) -> u64 {
+        match self.config_version {
+            Some(x) => x + 1,
+            None => 1,
+        }
+    }
+}
+
+impl<'a> TryFrom<&'a StringMap<'a>> for Totem {
+    type Error = ImlAgentError;
+
+    fn try_from(x: &'a StringMap<'a>) -> Result<Self, Self::Error> {
+        Ok(Totem {
+            config_version: x
+                .get("config_version (u64)")
+                .map(|x| x.parse())
+                .transpose()?,
+            cluster_name: (*try_get(x, "cluster_name (str)")?).to_string(),
+            token: try_get(x, "token (u32)")?.parse()?,
+        })
+    }
+}
+
+#[derive(Debug, Eq, PartialEq, Ord, PartialOrd)]
+pub(crate) struct Interface {
+    ringnumber: u32,
+    bindnetaddr: String,
+    mcastaddr: Option<String>,
+    broadcast: Option<String>,
+    mcastport: u16,
+}
+
+impl<'a> TryFrom<(&'a str, StringMap<'a>)> for Interface {
+    type Error = ImlAgentError;
+
+    fn try_from((ringnumber, x): (&'a str, StringMap<'a>)) -> Result<Self, Self::Error> {
+        Ok(Interface {
+            ringnumber: ringnumber.parse()?,
+            bindnetaddr: (*try_get(&x, "bindnetaddr (str)")?).to_string(),
+            mcastaddr: x.get("mcastaddr (str)").map(|x| (*x).to_string()),
+            broadcast: x.get("broadcast (str)").map(|x| (*x).to_string()),
+            mcastport: try_get(&x, "mcastport (u16)")?.parse()?,
+        })
+    }
+}
+
+/// View the current totem and interface information from `corosync-cmapctl`.
+async fn corosync_cmapctl() -> Result<(Totem, Vec<Interface>), ImlAgentError> {
+    let x = Command::new("corosync-cmapctl").checked_output().await?;
```

Comment:
If we only care about the cmap sections that appear in the config: 
```corosync-cmapctl totem logging quorum nodelist qb resources```
(though I've never seen a deployment with the last two)

Reply:
can just be `corosync-cmapctl totem` for now I think.